core/arch/mangle_shared.c

```diff
@@ -777,12 +777,58 @@ mangle_syscall_code(dcontext_t *dcontext, fragment_t *f, byte *pc, bool skip)
 static bool
 mangle_rseq(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr, instr_t *next_instr)
 {
+    int i;
     app_pc pc = get_app_instr_xl8(instr);
     app_pc start, end, handler;
-    if (!rseq_get_region_info(pc, &start, &end, &handler)) {
+    bool *reg_used;
+    int reg_used_size;
+    reg_id_t scratch_reg;
+    if (!rseq_get_region_info(pc, &start, &end, &handler, &reg_used, &reg_used_size,
+                              (ushort *)&scratch_reg)) {
         ASSERT_NOT_REACHED(); /* Caller was supposed to check for overlap */
         return false;
     }
+    /* We need to know the type of register so we can't completely abstract this. */
+    ASSERT(reg_used_size == DR_NUM_GPR_REGS);
```

Comment:
This seems to be an array of constant size in rseq_region_t. Do we need to return its size here, maybe remove and always use DR_NUM_PGR_REGS instead?

Reply:
rseq_region_t is hidden behind an abstraction barrer.  Since the struct is not exposed, the user here does not know the array size.  But given that it does need to know it's an array of just GPR's in order to preserve the registers, maybe the size could be just documented in the getter docs instead of returned.

Reply:
I don't know, returning the size still seems cleaner.

Reply:
I would say yes, if it may change in future (like non-GPR), not if it's fixed. Up to you, resolving.